### ISSUE TITLE
NCDatasets should fail when trying to access a nonexisting index.

### ISSUE THREAD

**Issue by aramirezreyes** [Alexander-Barth/NCDatasets.jl#30]
Hi:

When trying to read beyond the possible indices of a multifile dataset, NCDatasets doesn't fail, instead returns the array with some random data.

```
file2d = somelistoffiles
ds2d                = Dataset(file2d)
for (varname,var) in ds2d
    @show (varname,size(var))
end

```
returns: (varname, size(var)) = ("Prec", (512, 512, 2400))

If now I do

`Prec                   = ds2d["Prec"].var[:,:,2500]`

I get an array 512×512 Array{Float32,2} with some random information.

I would expect this to fail. Is that the expected behavior?

**Comment by Alexander-Barth** [comment]
Thank you reporting this issue. It is now fixed in the master version. Bounding errors are now thrown the cases you mentioned.

**Comment by Alexander-Barth** [comment]
For your information, this is fixed in the version 0.9.2.

**Comment by aramirezreyes** [comment]
This is great, thanks!
